### PR TITLE
Feat/rdf package

### PR DESCRIPTION
**Pull request by amimart**
The intention here is a preliminary work before going further in dataverse contract implementation. I propose through these changes to move RDF serde logic in its own package to allow it to be shared between contracts.

## Details

The `okp4-rdf` package has been thus introduced and used as dependencies by the `cognitarium` smart contract.

I also took the opportunity to implement RDF canonicalization as [described by the W3C](https://www.w3.org/TR/rdf-canon), as it'll be needed to verify proofs in verifiable credentials and presentations.

## Summary by CodeRabbit

- **New Features**
  - Introduced "RDF" package for enhanced management of RDF data.

- **Enhancements**
  - Streamlined RDF data handling by integrating new RDF normalization and serialization modules.

- **Documentation**
  - Updated README to introduce the new "RDF" package functionalities.

- **Refactor**
  - Replaced internal RDF processing with usage of the new `okp4_rdf` package to improve efficiency and maintainability.

- **Chores**
  - Removed unused code and imports to tidy up the codebase.

- **Bug Fixes**
  - Adjusted RDF data format handling to ensure compatibility and correct functionality.

### CODE REVIEW

**Review by ccamel — APPROVED** [review]
Awesome! :heart_eyes:

File: packages/okp4-rdf/src/normalize.rs [hunk 1]
```diff
@@ -0,0 +1,756 @@
...
+                });
+            });
+
+            let hash = Self::serialize(&replacements);
+            self.hash_to_blank_nodes
+                .entry(hash.clone())
+                .and_modify(|v| v.push(target.clone()))
+                .or_insert(vec![target.clone()]);
+            self.blank_node_to_hash.insert(target.clone(), hash);
+        }
```
[the remark this answers is not in the document]
Comment (amimart) [reply packages/okp4-rdf/src/normalize.rs:127]: The cloning of quads is necessary as we need to update them keeping the original data intact. And this doesn't bring too much concerns regarding performance in my sense as the quads, in their structure contains only references.

File: packages/okp4-rdf/src/normalize.rs [hunk 2]
```diff
@@ -0,0 +1,756 @@
...
+
+impl<T: Clone> Iterator for PermutationsIter<T> {
+    type Item = Vec<T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let crt = self.next.clone()?;
+        self.next = self.permute();
+        Some(crt)
+    }
+}
```
[the remark this answers is not in the document]
Comment (amimart) [reply packages/okp4-rdf/src/normalize.rs:489]: Fair point, will do :)

File: packages/okp4-rdf/src/normalize.rs [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+
+            hash_path_list.sort_by(|left, right| left.0.cmp(&right.0));
+            for (_, issuer) in hash_path_list {
+                for node in issuer.issue_log {
+                    self.canonical_issuer.get_or_issue(node.clone());
+                }
+            }
+        }
+
+        Ok(())
```
[the remark this answers is not in the document]
Comment (amimart) [reply packages/okp4-rdf/src/normalize.rs]: I see it difficult to split

File: packages/okp4-rdf/src/normalize.rs [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+            }
+
+            hasher.update(chosen_path.as_str());
+        }
+
+        Ok((
+            base16ct::lower::encode_string(&hasher.finalize()),
+            chosen_issuer,
+        ))
+    }
```
[the remark this answers is not in the document]
Comment (amimart) [reply packages/okp4-rdf/src/normalize.rs]: Seems difficult to split

File: packages/okp4-rdf/src/normalize.rs [hunk 5]
```diff
@@ -0,0 +1,756 @@
...
+                },
+            )
+            .collect();
+
+        for (hash, node) in unique_nodes? {
+            self.hash_to_blank_nodes.remove(&hash);
+            self.canonical_issuer.get_or_issue(node.clone());
+        }
+
+        Ok(())
```
[the remark this answers is not in the document]
Comment (amimart) [reply packages/okp4-rdf/src/normalize.rs]: Good point, I think it's better now

**Pull request merged**